implementation/src/main/java/io/smallrye/config/ConfigValueConfigSourceInterceptor.java

```diff
@@ -0,0 +1,42 @@
+package io.smallrye.config;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Pattern;
+
+import javax.annotation.Priority;
+
+import org.jboss.logging.Logger;
+
+@Priority(100)
+public class ConfigValueConfigSourceInterceptor implements ConfigSourceInterceptor {
+    private static final Logger LOG = Logger.getLogger("io.smallrye.config");
+    private static final String EXCLUDE_KEYS_PATTERN_KEY = "smallrye.log.exclude.keys.pattern";
+
+    private AtomicReference<Pattern> pattern = new AtomicReference<>();
+
+    @Override
+    public ConfigValue getValue(final ConfigSourceInterceptorContext context, final String name) {
+        ensureInitPattern(context);
+
+        final ConfigValue configValue = context.proceed(name);
+        if (!pattern.get().matcher(configValue.getName()).matches()) {
```

Comment:
I don't think a pattern is sufficient to determine if a key should be logged.  In Quarkus for example, assembling such a pattern is going to be fairly complex (particularly for a use case such as this one).  Given we want to solve the masking issue, why don't we remove this interceptor (for now) and reintroduce it once masking is solved?

Reply:
Ok. I'll remove this, but keep everything else (it is going to be needed anyway in the future) and we focus on #269